manifests/init.pp

```diff
@@ -87,4 +91,31 @@
       before  => File['satellite_config_yaml'],
     }
   }
+
+  if $trusted_external_command {
+    file { '/etc/puppetlabs/puppet/trusted-external-commands':
+      ensure  => directory,
+      owner   => 'pe-puppet',
+      group   => 'pe-puppet',
+      mode    => '0755',
+    }
+
+    ini_setting { 'satellite-trusted_external_command':
+      ensure  => present,
+      path    => '/etc/puppetlabs/puppet/puppet.conf',
+      section => 'master',
+      setting => 'trusted_external_command',
+      value   => '/etc/puppetlabs/puppet/trusted-external-commands',
+      notify  => Service['pe-puppetserver'],
+    }
+
+    file { '/etc/puppetlabs/puppet/trusted-external-commands/satellite':
+      ensure  => file,
+      owner   => 'pe-puppet',
+      group   => 'pe-puppet',
+      mode    => '0755',
+      source  => 'puppet:///modules/satellite_pe_tools/satellite',
```

Comment:
I've been getting using to using `content => file("${module_name}/satellite")` so it's inlined in the catalog.

Reply:
Curious, what would be the benefit of inlining this kind of file?

My instinct is to avoid inlining, since it's a static non-templated file with a fair bit of content to it. Static catalogs already inlines the checksum information, so the agent will only request a copy from the server in a separate API call if it _knows_ that it needs to replace it; most agent runs will read the inlined checksum, see that it matches, and proceed. Inlining the actual content would make the catalog larger, but not provide runtime performance improvements for the 99% case.

Is there another reason why inlining might be good here, or in general?

Reply:
Some of this may be outdated but back in the day it was faster for small files. Source means a HTTP GET which does generate some additional load. There is (by now) an ancient blog which benchmarked it and found that up to some size (IIRC 4k) it was faster to inline it. You may be right that by now puppetserver learned optimizations that remove the benefit.

To me now it mostly means that catalog compilation checks if the path is actually present. That means that rspec-puppet can ensure the source is available rather than checking it at runtime.

Another small thing may be that a cached catalog can be applied which could happen if the server is offline. I'm not sure how practical that is.

There is some downside: if you support `ensure => absent` the `content => file()` is still executed which is technically redundant.

Overall I like the pattern a lot more and rarely use `source =>` at all.

Reply:
Yeah, there is no longer any additional load from `source` as long as you have static catalogs enabled (by default in PE).

The rspec-puppet utility is interesting. I haven't thought about that one before.

My preference is still to use `source` when appropriate in order to keep the catalog smaller. But preferences like that are the module maintainer's call, and if as the maintainer you get a utility value out of using `content` instead, that's fine by me. 👍